File: src/main/java/com/puhovin/intellijplugin/twm/action/ApplyPreferredAvailabilitiesAction.java
package com.puhovin.intellijplugin.twm.action;

import com.puhovin.intellijplugin.twm.ToolWindowManagerService;
import com.puhovin.intellijplugin.twm.model.ToolWindowPreference;
import org.jetbrains.annotations.NotNull;

import java.util.List;

public class ApplyPreferredAvailabilitiesAction extends AbstractApplyAvailabilitiesAction {

    @Override
    @NotNull
    protected List<ToolWindowPreference> getPreferencesToApply(@NotNull ToolWindowManagerService toolWindowManagerProjectComponent) {
        return toolWindowManagerProjectComponent.getPreferredAvailabilities();
    }
}

File: src/main/kotlin/com/puhovin/intellijplugin/twm/action/ApplyPreferredAvailabilitiesAction.kt
package com.puhovin.intellijplugin.twm.action

import com.puhovin.intellijplugin.twm.core.ToolWindowManagerDispatcher
import com.puhovin.intellijplugin.twm.model.ToolWindowPreference
import org.jetbrains.annotations.NotNull

/**
 * Action to apply preferred tool window availabilities.
 */
class ApplyPreferredAvailabilitiesAction : AbstractApplyAvailabilitiesAction() {

    /**
     * Gets the preferred tool window preferences to apply.
     *
     * @param dispatcher The tool window manager dispatcher
     * @return A list of tool window preferences to apply
     */
    @NotNull
    override fun getPreferencesToApply(@NotNull dispatcher: ToolWindowManagerDispatcher): List<ToolWindowPreference> {
        return dispatcher.getCurrentAvailabilityToolWindows()
    }
}
